api/logger.py

```diff
@@ -0,0 +1,35 @@
+import logging
+from api.utils import running_local
+
+
+class DefaultLogger(object):
+  def __init__(self, is_running_local):
+    self.logger = logging.getLogger("pcluster-manager")
+    if is_running_local:
+      handler = logging.StreamHandler()
+      handler.setFormatter(logging.Formatter("%(asctime)s - %(levelname)s - %(message)s"))
+      self.logger.addHandler(handler)
+      self.logger.setLevel(logging.DEBUG)
+    else:
+      self.logger.setLevel(logging.INFO)
```

Comment:
issue(non-blocking): if no formatter is defined for prod, python would default to this: `"%(levelname)s:%(name)s:%(message)s` (as you can see [here](https://hg.python.org/cpython/file/5c4ca109af1c/Lib/logging/__init__.py#l1640))
Is this the desired formatter for prod deployment?

Reply:
Python logger in AWS Lambda has a preset format, which has the following structure:

`%(asctime)s - @requestid - %(level)s - %(message)s `

The available options were:
* Keep this format as is, avoiding to add another handler
* On logger configuration, remove all the existing handlers and attach our custom handler

Reply:
Since we have the `extra` parameter, does it make sense to leverage it?

For instance we will have something like this:

```
logger.info("Message", extra={"source": "backend"})

Reply:
In order to use that `extra` param you would need either a custom formatter or a json formatter to configure the logging module with.
You could use it if you know that certain parameters will always be provided and include it in the format string like, for example, if you know you always have a "client_ip" extra parameter, you could log it including
`%(client_ip)s` in your format string.

Reply:
The formatter is configured in the following way:

```
"%(asctime)s - %(levelname)s - %(message)s
```

Where `message` is a JSON which can be queried with CloudWatch Insights or parsed in any other way.